src/functions.js

```diff
@@ -121,7 +120,26 @@ var _ = Mavo.Functions = {
 			return n && n > 1? [arr] : arr;
 		}
 
-		return n && n > 1? arr.slice(0, n) : arr[0];
+		if (n >= 0) {
+			var ret = [];
+			n = Math.floor(n);
+
+			for (var i = 0; i<arr.length && ret.length<n; i++) {
+				if (Mavo.value(arr[i]) !== null) {
+					ret.push(arr[i]);
+				}
+			}
+
+			return ret;
+		} 
+		else if (n < 0) {
+			return _.last(Math.abs(n), arr);
+		}
+		else {
+			for (var i = 0; i<arr.length && Mavo.value(arr[i]) === null; i++) {}
+			return arr[i];
```

Comment:
A comment would be useful here, otherwise the reader has to read the whole code to understand that the else case means n is undefined.

Also why handle this separately and reimplement the searching algorithm? Isn't this basically the same as `first(1, arr)` except we return `ret[0]` (or null if ret is empty)?